src/ws/types.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * All the type definitions should be moved to @graasp/sdk
+ */
+
+import { AppAction, AppData, AppSetting } from '@graasp/sdk';
+
+export type AppOperations = 'post' | 'patch' | 'delete';
+
+export type AppEventKinds = 'app-data' | 'app-settings' | 'app-actions';
```

Comment:
Consider exporting the constant strings themselves to consume them where previously marked. Maybe in SDK.

Reply:
> Consider exporting the constant strings themselves to consume them where previously marked. Maybe in SDK.

And you can either put then in an `enum` and get this union type with 
```
`${YourEnum}`
``` 
or use a const object and use the `UnionOfConst` generic type from sdk to get the same.

Example:
```ts
enum AppEvent {
  AppData = "app-data",
  AppAction = "app-action",
  AppSetting = "app-setting",
}
type AppEventKind = `${AppEvent}` // resolves to "app-data" | "app-action" | "app-setting"

// OR
import { UnionOfConst } from "@graasp/sdk";

const AppEvent = {
  AppData: " app-data",
  AppAction: "app-action",
  AppSetting: "app-setting",
} as const;
type AppEventKind = UnionOfConst<typeof AppEvent> // resolves to "app-data" | "app-action" | "app-setting"

```